File: RPi Experiments/RPi2-1/C/01_blinkingLed.c
#include<wiringPi.h>
#include<stdio.h>

#define ledPin 0 //wiringPi pin number for the LED

int main(void)
{
	if(wiringPiSetup() == -1)
	{
		printf("Setup wiring Pi failed!"); //when initialize wiring failed,print messageto screen
		return 1;
	}
	
	pinMode(ledPin,OUTPUT); //initialize the LED pin as output
	
	while(1)
	{
		digitalWrite(ledPin,HIGH); //turn on the LED
		printf("Led ON...\n");
		delay(500); //wait for half a second
		digitalWrite(ledPin,LOW); //turn off the LED
		printf("Led OFF...\n"); 
		delay(500); //wait for half a second
	}
	
	return 0;
}

File: RPi Experiments/RPi2-1/Python/01_blinkingLed.py
import RPi.GPIO as GPIO
import time

ledPin = 11 # physical pin number for the LED

def setup():
    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD) # use GPIO by physical location
    GPIO.setup(ledPin,GPIO.OUT) # initialize the LED pin as output
    GPIO.output(ledPin,GPIO.HIGH) # turn off led

def loop():
	while True:
		print "Led ON..."
		GPIO.output(ledPin,GPIO.LOW) # turn on led
		time.sleep(0.5) # wait for half a second
		print "Led OFF..."
		GPIO.output(ledPin,GPIO.HIGH) # turn off led
		time.sleep(0.5) # wait for half a second

def destroy():
	GPIO.output(ledPin,GPIO.HIGH) # turn off led
	GPIO.cleanup() # release resource

if __name__ == '__main__': # program start from here
	try:
		setup()
		loop()

	except KeyboardInterrupt: # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()